src/utils/telemetry.ts

```diff
@@ -207,3 +225,43 @@ export const consoleTelemetryExporter: TelemetryExporter = {
     console.debug('Telemetry Count', count);
   },
 };
+
+/**
+ * Telemetry entry point to initialize the singleton {@link TelemetryProvider} or
+ * get the current {@link TelemetryProvider}
+ */
+export class Telemetry {
+  private static instance: TelemetryProvider | undefined;
+
+  /**
+   * Gets the singleton {@link TelemetryProvider} configured for the Data Layer Observer instance.
+   * Initializes the singleton {@link TelemetryProvider} on first call
+   *
+   * @param provider The {@link TelemetryProvider} to initialize; only effective on first call
+   * @param exporter The {@link TelemetryExporter} to use with the default Telemetry Provider; only
+   * used if no {@link TelemetryProvider} is given and only effective on first call
+   */
+  static getInstance(provider?: TelemetryProvider, exporter?: TelemetryExporter): TelemetryProvider {
+    if (Telemetry.instance) {
+      return Telemetry.instance;
+    }
+
+    if (provider) {
+      Telemetry.instance = provider;
+    } else if (exporter) {
+      Telemetry.instance = new DefaultTelemetryProvider(exporter);
+    } else {
+      Telemetry.instance = new DefaultTelemetryProvider(consoleTelemetryExporter);
+    }
+
+    return Telemetry.instance;
+  }
+
+  /**
+   * Clears the initialized telemetry provider allowing telemetry to be reinitialized. Used
+   * within telemetry intialization tests to achieve a clean starting state
+   */
+  static reset(): void {
```

Comment:
I'm not enthusiastic about adding a `reset` method strictly for tests. This is one of the challenges of following the singleton pattern we use for logging. For logging, we allow directly modifying the appender and log levels on the singleton logging instance. Since telemetry configuration should not be changed after initializing other than within tests, I opted for a `reset` method instead of exposing telemetry state for manipulation (e.g. changing the provider and/or exporter after initializing).

Reply:
`setInstance` (from our prior thread) may solve this problem?